superset/db_engine_specs/trino.py

```diff
@@ -127,14 +129,42 @@ def get_tracking_url(cls, cursor: "Cursor") -> Optional[str]:
         return None
 
     @classmethod
-    def handle_cursor(cls, cursor: "Cursor", query: Query, session: Session) -> None:
-        """Updates progress information"""
+    def handle_cursor(cls, cursor: Cursor, query: Query, session: Session) -> None:
         tracking_url = cls.get_tracking_url(cursor)
         if tracking_url:
             query.tracking_url = tracking_url
-            session.commit()
+
+        # Adds the executed query id to the extra payload so the query can be cancelled
+        query.set_extra_json_key("cancel_query", cursor.stats["queryId"])
+
+        session.commit()
         BaseEngineSpec.handle_cursor(cursor=cursor, query=query, session=session)
 
+    @classmethod
+    def has_implicit_cancel(cls) -> bool:
+        return False
+
+    @classmethod
+    def cancel_query(cls, cursor: Any, query: Query, cancel_query_id: str) -> bool:
+        """
+        Cancel query in the underlying database.
+
+        :param cursor: New cursor instance to the db of the query
+        :param query: Query instance
+        :param cancel_query_id: Trino `queryId`
+        :return: True if query cancelled successfully, False otherwise
+        """
+        try:
+            cursor.execute(
+                f"CALL system.runtime.kill_query('{cancel_query_id}',"
+                "'Query cancelled by Superset')"
+            )
```

Comment:
Hi @villebro, I am not running Trino in my local. Is it same as "arrow arguments" ? because I found the kill SQL in the manual is 
```
CALL system.runtime.kill_query(query_id => '20151207_215727_00146_tx3nr', message => 'Using too many resources');
```

Reply:
> [trinodb/trino-python-client#195](https://github.com/trinodb/trino-python-client/pull/195) add cancel query on `cursor.close()` - release in trino driver 0.315
> 
> @villebro did u test it out. If it's work, I think it's better to just upgrade trino driver to 0.315

Thanks, I wasn't aware of this! Interesting - this was actually already tested on 0.315. But let me look into this more closely.

Reply:
Thanks for the heads up @zhaoyongjie. This appears to be a more explicit way of passing arguments and probably more future proof if the sig changes. I'll make the change and validate.

Reply:
@zhaoyongjie I can confirm that the more explicit syntax works equally well (pushed in latest commit).